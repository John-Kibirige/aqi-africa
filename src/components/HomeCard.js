import React from 'react';
import rightArrow from '../assets/right-arrow.svg';

const HomeCard = (props) => {
  const { cardImage, country, aqIndex } = props;
  return (
    <div>
      <img src={rightArrow} alt="right arrow" />
      <img src={cardImage} alt="country map" />
      <h1>{country}</h1>
      <p>{aqIndex}</p>
    </div>
  );
};

export default HomeCard;
