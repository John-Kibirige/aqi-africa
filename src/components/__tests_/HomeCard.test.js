import React from 'react';
import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import HomeCard from '../HomeCard';

it('Test to see if the home card renders well ', () => {
  render(<HomeCard imageUrl="url" country="Uganda" aqIndex={3} />);
  const renderedText = screen.getByText(/Air Quality Index/i);
  expect(renderedText).toBeInTheDocument();
});

it('Create sapshot for home card component ', () => {
  const tree = renderer
    .create(<HomeCard imageUrl="url" country="Uganda" aqIndex={3} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
