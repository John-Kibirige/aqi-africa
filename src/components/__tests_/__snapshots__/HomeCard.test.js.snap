// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create sapshot for home card component  1`] = `
<div
  className="home-card rounded flex-col flex justify-between"
>
  <img
    alt="right arrow"
    className="arrow w-7 block ml-auto m-4"
    src="right-arrow.svg"
  />
  <img
    alt="country map"
    className="map w-full h-32 sm:h-40 object-contain"
    src="url"
  />
  <div
    className="home-card-text text-right"
  >
    <h1
      className="home-card-title uppercase text-xl text-pink-500 sm:text-pink-700 sm:mt-2 sm:mr-3"
    >
      Uganda
    </h1>
    <p
      className="text-center sm:text-xl"
    >
      Air Quality Index
      <span
        className="bg-pink-500 inline-block text-white p-1 px-3 m-1 font-bold  rounded-3xl"
      >
         
      </span>
    </p>
  </div>
</div>
`;
